### PR TITLE
Bug 1246774 - Redesign performance alerts management UI

### PR DESCRIPTION
**Pull request by wlach**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1294)

### CODE REVIEW

File: treeherder/perf/models.py [hunk 1]
```diff
@@ -100,14 +101,86 @@ class PerformanceAlertSummary(models.Model):
...
+    def update_status(self):
+        autodetermined_status = self.autodetermine_status()
+        if autodetermined_status != self.status:
+            self.status = autodetermined_status
+            self.save()
+
+    def autodetermine_status(self):
+        alerts = (PerformanceAlert.objects.filter(summary=self) |
+                  PerformanceAlert.objects.filter(related_summary=self))
+        print alerts
```
Comment (jmaher) [inline treeherder/perf/models.py]: is this print statement supposed to be there?

File: treeherder/perf/models.py [hunk 2]
```diff
@@ -129,25 +202,23 @@ class PerformanceAlert(models.Model):
...
 
     STATUSES = ((UNTRIAGED, 'Untriaged'),
+                (DOWNSTREAM, 'Downstream'),
+                (REASSIGNED, 'Reassigned'),
                 (INVALID, 'Invalid'),
-                (WONTFIX, 'Won\'t fix'),
-                (RESOLVED, 'Resolved'),
-                (INVESTIGATING, 'Investigating'),
-                (DUPLICATE, 'Duplicate'))
+                (ACKNOWLEDGED, 'Acknowledged'))
```
Comment (jmaher) [inline treeherder/perf/models.py]: why are these different from the statuses above?  do we have different status for summary vs alert?

File: ui/js/controllers/perf/alerts.js [hunk 3]
```diff
@@ -120,10 +432,15 @@ perf.controller('AlertsCtrl', [
             _.forEach($scope.alertSummaries, function(alertSummary) {
                 _.forEach(alertSummary.alerts, function(alert) {
                     // only show alert if it passes all filter criteria
+                    // also hide downstream alerts that are not directly related
+                    // to this summary (FIXME: maybe show something underneath
+                    // the alert like "XX downstream alerts not shown" and provide
+                    // a button to disclose them?)
```
Comment (jmaher) [inline ui/js/controllers/perf/alerts.js]: yes, please make this a new bug, that would be useful

File: ui/js/values.js [hunk 4]
```diff
@@ -177,13 +177,23 @@ treeherder.value("phTimeRanges", [
...
+    WONTFIX: { id: 6, text: "wontfix" },
+    RESOLVED: { id: 7, text: "resolved" }
+});
+
+treeherder.value("phAlertStatusMap", {
+    UNTRIAGED: { id: 0, text: "untriaged" },
+    DOWNSTREAM: { id: 1, text: "downstream" },
+    REASSIGNED: { id: 2, text: "reassigned" },
+    INVALID: { id: 3, text: "invalid" },
+    ACKNOWLEDGED: { id: 4, text: "acknowledged" }
```
Comment (jmaher) [inline ui/js/values.js]: why are alert summary and alert status different sets of values?  I assume alert summary is related to a bug, and alert status could be filtered out into other things like downstream or invalid.

File: treeherder/perf/models.py (hunk 1 above)
Comment (wlach) [inline treeherder/perf/models.py]: lolno

File: treeherder/perf/models.py (hunk 2 above)
Comment (wlach) [inline treeherder/perf/models.py]: Yes. They mean different things, and it is allowed to have a different combination of alert statuses within a summary (depending on what those statuses are).

Currently this isn't documented, it's just based on my intuition of what makes sense. We'll probably be doing some tweaking as we go... once we're happy, we can write some documentation.

File: ui/js/controllers/perf/alerts.js (hunk 3 above)
Comment (wlach) [inline ui/js/controllers/perf/alerts.js]: Will do.

File: ui/js/values.js (hunk 4 above)
Comment (wlach) [inline ui/js/values.js]: See answer above. A "summary" refers to a group of alerts.

File: ui/js/controllers/perf/alerts.js (hunk 3 above)
Comment (wlach) [inline ui/js/controllers/perf/alerts.js]: https://bugzilla.mozilla.org/show_bug.cgi?id=1247028

**Pull request merged**